### ISSUE TITLE
[FEAT]: Multiple LLM Preferences

### ISSUE THREAD

**Issue by wcwong** [Mintplex-Labs/anything-llm#1567]
### What would you like to see?

I would like to be able to instantiate multiple instances of a  LLM provider. For example, with Azure I can create multiple endpoints (in different regions or different models).  It'd be nice not to have to continually change the one option.

**Comment by wallartup** [comment]
Check workspace settings and you can change models per workspace. LM studio is however missing from the list.

**Comment by wcwong** [comment]
> Check workspace settings and you can change models per workspace. LM studio is however missing from the list.

Yes, once I configure a model, I can assign that model to a workspace. However, once I configure, for example, the Generic OpenAI model, I am not able to reconfigure it per workspace. 

The OpenAI configuration has a separate drop down so using the same API key I can have different models per workspace but it doesn't appear to allow me to have different API keys for different workspaces.

**Comment by mrdjohnson** [comment]
> Check workspace settings and you can change models per workspace. LM studio is however missing from the list.

> I found this comment in the codebase actually: `Some providers can only be associated with a single model.`

I'm pretty positive thats an error actually, double checking now

**Comment by johannoriel** [comment]
same here, I would like to be able to use different key API per workspace (each user can already have it's own API key different from the other user)

**Comment by manzke** [comment]
the issue is that generic open ai and others are using the variables. when you create a workspace and configure it, it will write the settings into the .env file.
when you create another workspace and set different url for example, it will overwrite it. llm settings should be stored in the database and not the .env file. 
i would also introduce the idea of llm instances, so you can configure multiple llm instances with different settings and just apply one of it to a workspace.
